Patch-Source: https://github.com/falconry/falcon/commit/f5920fea1cea39fb2a1ce3fdee3ea335532f5ac4
From f5920fea1cea39fb2a1ce3fdee3ea335532f5ac4 Mon Sep 17 00:00:00 2001
From: Vytautas Liuolia <vytautas.liuolia@gmail.com>
Date: Tue, 21 Jan 2025 20:30:05 +0100
Subject: [PATCH] chore(requirements): unpin some test dependencies (#2430)

* chore(requirements): unpin some test dependencies

* chore(tests): unpin `websockets` 14.2
---
 .github/workflows/tests.yaml    |  2 ++
 requirements/tests              | 13 ++++---------
 tests/asgi/test_asgi_servers.py |  6 +++++-
 3 files changed, 11 insertions(+), 10 deletions(-)

diff --git a/tests/asgi/test_asgi_servers.py b/tests/asgi/test_asgi_servers.py
index 044d46a38..35372ec28 100644
--- a/tests/asgi/test_asgi_servers.py
+++ b/tests/asgi/test_asgi_servers.py
@@ -330,10 +330,14 @@ async def test_select_subprotocol_unknown(self, server_url_events_ws):
         except websockets.exceptions.NegotiationError as ex:
             assert 'unsupported subprotocol: xmpp' in str(ex)
 
-        # Daphne, Hypercorn
+        # Daphne, Hypercorn with websockets<14.2
         except EOFError:
             pass
 
+        # Daphne, Hypercorn with websockets>=14.2
+        except websockets.exceptions.InvalidMessage as ex:
+            assert isinstance(ex.__cause__, EOFError)
+
     # NOTE(kgriffs): When executing this test under pytest with the -s
     #   argument, one should be able to see the message
     #   "on_websocket:WebSocketDisconnected" printed to the console. I have
